common/threads.h

```diff
@@ -16,8 +16,10 @@
 
 #if defined(__APPLE__)
 #include <AvailabilityMacros.h>
-#if (MAC_OS_X_VERSION_MIN_REQUIRED > 1050) && !defined(__ppc__)
+#include <TargetConditionals.h>
```

Comment:
I am not sure `TargetConditionals.h` exists in every MacOS SDK. This has to be verified, otherwise it will break the build.

Reply:
Yes, the `TargetConditionals.h` exists on all modern apple platforms